cmake/modules/extensions.cmake

```
@@ -5532,3 +5520,35 @@ function(add_llext_command)
     COMMAND_EXPAND_LISTS
   )
 endfunction()
+
+# 7.3 llext helper functions
+
+# Usage:
+#   llext_filter_zephyr_flags(<filter> <flags> <outvar>)
+#
+# Filter out flags from a list of flags. The filter is a list of regular
+# expressions that will be used to exclude flags from the input list.
+#
+# The resulting list of flags will be stored in the variable <outvar>.
+#
+# Example:
+#   llext_filter_zephyr_flags(LLEXT_REMOVE_FLAGS zephyr_flags zephyr_filtered_flags)
+#
+function(llext_filter_zephyr_flags filter flags outvar)
+  list(TRANSFORM ${filter}
+       REPLACE "(.+)" "^\\1$"
+       OUTPUT_VARIABLE llext_remove_flags_regexp
+  )
+  string(REPLACE ";" "|" llext_remove_flags_regexp "${llext_remove_flags_regexp}")
```

I know this is moved code, but it caught my attention now.

This form of replacement is actually risky because it also replaces an escaped `\;` with a `\|` even though an escaped `;` is not separating items in a CMake list.

---

Not sure why I originally went down that road. Perhaps this alternative is better:
```suggestion
  list(JOIN llext_remove_flags_regexp "|" llext_remove_flags_regexp)
```

---

This can be done in a follow up imho

---

Yep - indeed, as this bug is present on current code, I've created another [PR](https://github.com/zephyrproject-rtos/zephyr/pull/72437) to tackle that - as this one can still drag for some time.